packages/material-ui/src/Popper/Popper.js

```diff
@@ -44,15 +46,23 @@ const Popper = React.forwardRef(function Popper(props, ref) {
     modifiers,
     open,
     placement: placementProps = 'bottom',
-    popperOptions = {},
+    popperOptions = defaultPopperOptions,
+    popperRef: popperRefProp,
     transition = false,
     ...other
   } = props;
   const tooltipRef = React.useRef(null);
-  const popperRef = React.useRef();
+  const handleRef = useForkRef(tooltipRef, ref);
+
+  const popperRef = React.useRef(null);
+  const instance = React.useRef();
+  const handlePopperRef = useForkRef(popperRef, popperRefProp);
+  React.useEffect(() => {
```

Comment:
this needs a layout effect just like the other useEventCallback

Reply:
I don't follow, this effect is the first to run. What problem do you envision?

Reply:
Ok, I see one.

Reply:
The default behavior is layout effect for aggressively memoized callbacks. Why did you deviate?

On another note: This makes fork ref useless since a change in refs won't give them current popper instances.

Reply:
Well no, false alarm. For one moment, I thought that the Portal onRendered could be a problem, it's not.

Reply:
Correct. I believe that we should go back to the initial approach. The previous approach had the disadvantage of creating a new popper.js instance at each render (not hiding the element as you said). Most people will use useState or useRef to collect the element. I believe it's not a case we should optimize for. thoughts?